### PR TITLE
closes #200: make scadnano pitch angle agree with oxDNA

### LINKED ISSUE #200: make scadnano pitch angle agree with oxDNA
See https://github.com/UC-Davis-molecular-computing/scadnano/issues/667.

Depending on how this is resolved, it may require nothing to be done in the Python package. We might resolve it by changing how pitch angles are interpreted in the web interface, in which case the oxDNA export code (in both the Python package and web interface) would remain the same.

### PR DESCRIPTION
**Pull request by DanielHader**
closes #200: make scadnano pitch angle agree with oxDNA

### CODE REVIEW

File: scadnano/scadnano.py [hunk 1]
```diff
@@ -6833,15 +6834,32 @@ def _oxdna_get_helix_vectors(design: Design, helix: Helix) -> Tuple[_OxdnaVector
...
+    pitch_axis = pitch_axis.rotate(-design.yaw_of_helix(helix), yaw_axis)
+    roll_axis = roll_axis.rotate(-design.yaw_of_helix(helix), yaw_axis)
+
+    # then the pitch rotation
+    yaw_axis = yaw_axis.rotate(-design.pitch_of_helix(helix), pitch_axis)
+    roll_axis = roll_axis.rotate(-design.pitch_of_helix(helix), pitch_axis)
+
+    # then the roll rotation
+    yaw_axis = yaw_axis.rotate(-design.roll_of_helix(helix), roll_axis)
+    pitch_axis = pitch_axis.rotate(-design.roll_of_helix(helix), roll_axis)
```
Comment (dave-doty) [inline scadnano/scadnano.py:6856]: This code seems suspicious to me. The variable `pitch_axis` is not used after this point, so if assigning to it is supposed to have some effect, it won't.

Also, it isn't what I was expecting. The example shown in https://github.com/UC-Davis-molecular-computing/scadnano/issues/667, which looks like this in scadnano:

image

now displays in oxDNA like this:

image

It should look like this (but with the blue axis pointing right and the green axis pointing down):

image

**Review by dave-doty — CHANGES_REQUESTED** [review]
Doesn't seem to work as we want. See other comment below.

It might be a good idea to make up some small unit tests to express intended behavior. I'm not quite sure what good unit tests would look like, since testing exact float-point positions of bases after the rotation seems brittle and difficult to figure out.

**Comment by DanielHader** [comment]
Oops, sorry about that! I accidentally switched the x and z axes which is
why it looked right, but in the wrong orientation. I just committed a fixed
version.

I also removed the last write to the pitch_axis variable which I left in
when I was making changes to the code originally. I thought I might use it
later so I updated it, but I never ended up using it.

Yeah, I'm not really sure what good unit tests would be for this, I could
check for floating point equality upto some small epsilon which should be
good enough to check against things like rotations being backwards.

Thanks,
Daniel

**Review by dave-doty — APPROVED** [review]
I implemented it in Dart and played with it a bit, and it seems to give expected results. It would be nice to have some unit tests, but like I said I'm not sure how to make them robust.

**Pull request merged**